src/components/header/Header.tsx

```diff
@@ -23,37 +26,85 @@ const StyledHeader = withProps<{ height: string }>(styled.header)`
     z-index: 100;
 `;
 
-const StyledLink = styled(Link)`
+const StyledLink = styled(RouterLink)`
     display: inline-block;
 `;
 
+const DataSetLink = Link.extend`
+    display: inline-block;
+    color: ${props => props.theme.colors.white};
+    border: 1px solid ${props => props.theme.colors.shade.medium};
+    border-radius: 0.125rem;
+    padding: 0.25rem 0.5rem;
+    align-self: center;
+    margin-left: 0.5rem;
+    transition: all 0.2s ease;
+
+    &:hover {
+        color: ${props => props.theme.colors.primary.light};
+        border-color: ${props => props.theme.colors.primary.light};
+    }
+`;
+
 const StyledImg = styled.img`
     margin-left: 1rem;
     height: 2.5rem;
     display: inline-block;
 `;
 
-interface Props {
+interface StateProps {
     user: UserReducer;
+    routing: { location: Location } | null;
+}
+
+interface DispatchProps {
     onLogOut: () => void;
 }
 
-const Header: SFC<Props> = ({ user, onLogOut }) => {
+type FullProps = StateProps & DispatchProps & RouteComponentProps<any>;
+
+const regex = new RegExp(`${ROUTE_PATHS.details}|${ROUTE_PATHS.edit}|${ROUTE_PATHS.search}`);
+const getDataSetFromRoute = (location: Location | null): { key: string; value: string } | null => {
+    if (location && location.pathname) {
```

Comment:
waarom parse je hier de route zelf? zit het niet in de params? if so, waarom niet?